pkg/runtime/serializer/json/json.go

```diff
@@ -119,8 +121,19 @@ func (s *Serializer) Decode(originalData []byte, gvk *unversioned.GroupVersionKi
 		typed, _, err := s.typer.ObjectKind(into)
 		switch {
 		case runtime.IsNotRegisteredError(err):
-			if err := codec.NewDecoderBytes(data, new(codec.JsonHandle)).Decode(into); err != nil {
-				return nil, actual, err
+			_, ok := into.(codec.Selfer)
```

Comment:
There is no easy way to tell if the Selfer interface is implemented by `into` or by `into.TypeMeta`. I'm giving up this effort.